### PR TITLE
use fedora.linux_system_roles.certificate role

### PR DESCRIPTION
**Pull request by richm**
The dependencies will be installed everywhere, so just use them.

### CODE REVIEW

**Comment by richm** [comment]
@martinpitt the downstream tests don't like the current code - I think there is no reason to do a git checkout of the certificate role when the latest is available in the fedora collection, plus the cockpit role now has an explicit dependency on the system roles collection, so might as well use it.

**Comment by richm** [comment]
[citest]

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -23,7 +23,16 @@ jobs:
           sudo apt update
           sudo apt install -y ansible
 
-      - name: Run test playbook
+      - name: Install dependencies
+        run: |
+          set -euo pipefail
+          for ff in meta/collection-requirements.yml tests/collection-requirements.yml; do
+            if [ -f "$ff" ]; then
```
Comment (martinpitt) [inline .github/workflows/integration-tests.yml:30]: Feels a little brittle/unnecessarily complicated -- we *know* that the first exists, and the second doesn't. But not a strong opinion.

**Review by martinpitt — DISMISSED** [review]
The effect of this is that lsr.cockpit git does not test any more against e.g. lsr.certificate git master, but against the latest released version. This makes it harder to co-develop API/features. But if these modules are released to galaxy often, that should be okay. I just want to make sure that you considered that?

Thanks!

**Comment by richm** [comment]
> The effect of this is that lsr.cockpit git does not test any more against e.g. lsr.certificate git master, but against the latest released version. This makes it harder to co-develop API/features.

Yes.  But how often does that happen?

> But if these modules are released to galaxy often, that should be okay.

We have greatly improved and speeded up the procedure for releasing roles to Galaxy.  And it is now mostly automated.

> I just want to make sure that you considered that?

Yes.  In the future, I'd also like to have a way that local role developers can work against a local clone copy of another role.

In practice, this happens very, very infrequently, so it's not a problem that needs to be solved in a general way at the present time.

> 
> Thanks!

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (richm) [reply .github/workflows/integration-tests.yml:30]: > Feels a little brittle/unnecessarily complicated -- we _know_ that the first exists, and the second doesn't. But not a strong opinion.

I believe it make this less brittle/more robust - https://linux-system-roles.github.io/documentation/role-requirements.html
```
Collections required at test-time will be specified in tests/collection-requirements.yml. Users will typically not need to know about this, but developers can use this with tox-lsr, the CI system, etc.
```
so if we do add test dependencies in the future, CI will "just work", and the developer doesn't have to know to change the github workflow file.

And - this is how all of our other CI works - https://github.com/linux-system-roles/tox-lsr/blob/main/src/tox_lsr/test_scripts/runqemu.py#L949 - and - https://gitlab.com/testing-farm/gluetool-modules/-/blob/main/gluetool_modules_framework/testing/test_scheduler_system_roles.py#L194 - so it makes cockpit integration testing consistent with other system roles integration testing.

**Comment by martinpitt** [comment]
>  But how often does that happen?

I don't know in general -- just that I did that as one of my first contributions, for updating the certificate role, fixing the selinux role, and then using these in cockpit. But you sound like this wouldn't happen very often. Anyway, you thought about this, that's all I wanted to know. Thanks!

**Pull request merged**